### DISCUSSION TITLE
How to update to specific unstable version in regexManager

### DISCUSSION THREAD

**Discussion by michutrain** [renovatebot/renovate#21238] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.63.1

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

7.21.7

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I want the regexManager to update this string from one unstable version to another, but only increment the version numbers and not the unstable suffix.
```
foo: 
 1.0.0-jre17
 1.0.0-jre17-xyz
 1.0.0-jre17-xyz-test
 1.0.1-jre17
 1.0.1-jre17-xyz
 1.0.1-jre17-xyz-test
```

Updating `foo` should go from `1.0.0-jre17` to `1.0.1-jre17`:
```groovy
fooVersionFallback: "1.0.0-jre17"
```
should become
```groovy
fooVersionFallback: "1.0.1-jre17"
```

### What you tried so far.

I've looked at https://docs.renovatebot.com/modules/manager/regex/ and https://docs.renovatebot.com/configuration-options/#extractversion, and if i'm understanding the documentation right, `extractVersionTemplate` should be what i'm looking for, but my regex seems to work fine on regex101 (https://regex101.com/r/Ly7O1x/1879). 
```js
...
  regexManagers: [
    {
      "fileMatch": ["^Jenkinsfile$"],
      "matchStrings": ["fooVersionFallback = \"(?<currentValue>.*)\""],
      "depNameTemplate": "foo",
      "datasourceTemplate": "artifactory",
      "registryUrlTemplate": "https://registry.com/artifactory/",
      "extractVersionTemplate": "^(?<version>\\d+\\.\\d+\\.\\d+\\-[a-zA-Z0-9]*).*$"
    }
  ],
  ...
  ```
The regexManager isn't being modified in the packageRules either.
Currently, instead of bumping to `1.0.1-jre17`, this bumps it to `1.0.0-jre17-xyz-test` instead.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: packageFiles with updates 
      "config": {
        "regex": [
           {
             "packageFile": "Jenkinsfile",
             "deps": [
               {
                 "depName": "foo",
                 "currentValue": "1.0.0-jre17",
                 "datasource": "artifactory",
                 "registryUrls": [
                   "https://registry.com/artifactory/"
                 ],
                 "replaceString": "fooVersionFallback = \"1.0.0-jre17\"",
                 "depIndex": 0,
                 "updates": [
                   {
                     "bucket": "non-major",
                     "newVersion": "1.0.0-jre17-xyz",
                     "newValue": "1.0.0-jre17-xyz",
                     "releaseTimestamp": "2023-03-07T00:32:00.000Z",
                     "newMajor": 18,
                     "newMinor": 0,
                     "updateType": "patch",
                     "branchName": "renovate/all-minor-patch"
                   }
                 ],
                 "warnings": [],
                 "versioning": "semver",
                 "registryUrl": "https://registry.com/artifactory",
                 "currentVersion": "1.0.0-jre17",
                 "isSingleVersion": true,
                 "fixedVersion": "1.0.0-jre17"
               }
             ],
             "matchStrings": ["fooVersionFallback = \"(?<currentValue>.*)\""],
             "depNameTemplate": "foo",
             "datasourceTemplate": "artifactory",
             "registryUrlTemplate": "https://registry.com/artifactory/"
           }
         ]
       }
```

</details>

**Comment by rarkins** [comment, marked as answer]
This might work if you use versioning=docker, because its default behavior is to ensure that the `-suffix` always matches.

**Comment by michutrain** [comment]
Yup, ended up using `versioning=loose` along with the regex for my implementation but by the sounds of it, `docker` would be even better. Thanks!

**Reply by rarkins** [reply]
loose might only work reliably if you are using extractVersion to enforce that the suffix is present. While with docker versioning you don't need that manual step as by default it ensures that the new version has the same suffix as the current version
